packages/eslint-plugin/src/rules/no-single-styles-array.ts

```diff
@@ -0,0 +1,70 @@
+import { ASTUtils, Selectors } from '@angular-eslint/utils';
+import type { TSESTree } from '@typescript-eslint/utils';
+import { createESLintRule } from '../utils/create-eslint-rule';
+
+type Options = [];
+export type MessageIds = 'noSingleStylesArray' | 'noSingleStyleUrl';
+export const RULE_NAME = 'no-single-styles-array';
+
+export default createESLintRule<Options, MessageIds>({
+  name: RULE_NAME,
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'Ensures component `styles`/`styleUrl` with `string` is used over `styles`/`styleUrls` when there is only a single string in the array',
+    },
+    fixable: 'code',
+    schema: [],
+    messages: {
+      noSingleStyleUrl:
+        'Use `styleUrl` over `styleUrls` for a single stylesheet',
+      noSingleStylesArray:
+        'Use a `string` over `[string]` for the `styles` property',
+    },
+  },
+  defaultOptions: [],
+  create(context) {
+    const {
+      COMPONENT_CLASS_DECORATOR,
+      LITERAL_OR_TEMPLATE_ELEMENT,
+      metadataProperty,
+    } = Selectors;
+    const singleArrayStringLiteral = `ArrayExpression:matches([elements.length=1]:has(${LITERAL_OR_TEMPLATE_ELEMENT}))`;
+    const singleStylesArrayExpression = `${COMPONENT_CLASS_DECORATOR} ${metadataProperty(
+      'styles',
+    )} > ${singleArrayStringLiteral}`;
+    const singleStyleUrlsProperty = `${COMPONENT_CLASS_DECORATOR} ${metadataProperty(
+      'styleUrls',
+    )}:has(${singleArrayStringLiteral})`;
+
+    return {
+      [singleStylesArrayExpression](node: TSESTree.ArrayExpression) {
+        context.report({
+          node,
+          messageId: 'noSingleStylesArray',
+          fix: (fixer) => {
+            const [el] = node.elements;
+            // TODO: Do we need the fix to apply for template strings?
```

Comment:
I think the fix will need to apply for template strings. Using backticks for quotes is perfectly valid and, as far as I can tell, the string is not considered a "literal" (it has a type of `TemplateLiteral` rather than `Literal`):

```ts
@Component({
    styleUrls: [`./widget.component.ts`]
}
```

Reply:
I'm not sure if getting the element's text from the source code is the preferred way, but this works for me:

```ts
fix: (fixer) => {
  const [el] = node.elements;
  if (el) {
    if (ASTUtils.isStringLiteral(el)) {
      return [fixer.replaceText(node, el.raw)];
    }
    if (ASTUtils.isTemplateLiteral(el)) {
      return [
        fixer.replaceText(
          node,
          `${context.getSourceCode().getText(el)}`,
        ),
      ];
    }
  }
  return [];
},
```